File: TP 1/Q15(DONE 99_)/tp1.java
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class tp1 {

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String word = "";

        String x1;
        String x2;
        String x3;
        String x4;

        while (!word.equals("FIM")) {
            word = br.readLine();
            if (isConsonant(word, 0, true) == true) {
                x1 = "SIM";
            } else {
                x1 = "NAO";
            }
            if (isVogal(word, 0, true) == true) {
                x2 = "SIM";
            } else {
                x2 = "NAO";
            }
            if (isInteger(word) == true) {
                x3 = "SIM";
            } else {
                x3 = "NAO";
            }
            if (isFloat(word) == true) {
                x4 = "SIM";
            } else {
                x4 = "NAO";
            }
            System.out.println(x1 + " " + x2 + " " + x3 + " " + x4 + " ");
        }
    }

    public static boolean isConsonant(String word, int control, boolean result) {
        if (control < word.length()) {
            if (!(word.charAt(control) != 'a' && word.charAt(control) != 'e' && word.charAt(control) != 'i'
                    && word.charAt(control) != 'o' && word.charAt(control) != 'u'
                    && word.charAt(control) != 'A' && word.charAt(control) != 'E' && word.charAt(control) != 'I'
                    && word.charAt(control) != 'O' && word.charAt(control) != 'U')) {
                result = false;
            }
            // System.out.println(
            // "ISCONSONANT Letra = " + word.charAt(control) + "/result = " + result +
            // "/control = " + control);
            result = isConsonant(word, control + 1, result);

        }

        return result;
    }

    public static boolean isVogal(String word, int control, boolean result) {
        if (control < word.length()) {
            if (word.charAt(control) != 'a' && word.charAt(control) != 'e' && word.charAt(control) != 'i'
                    && word.charAt(control) != 'o' && word.charAt(control) != 'u'
                    && word.charAt(control) != 'A' && word.charAt(control) != 'E' && word.charAt(control) != 'I'
                    && word.charAt(control) != 'O' && word.charAt(control) != 'U') {
                result = false;
            }
            // System.out.println(
            // "ISVOGAL Letra = " + word.charAt(control) + "/result = " + result + "/control
            // = " + control);
            result = isVogal(word, control + 1, result);

        }
        return result;
    }

    public static boolean isInteger(String string) {
        int intValue;

        // System.out.println(String.format("Parsing string: \"%s\"", string));

        try {
            intValue = Integer.parseInt(string);
            return true;
        } catch (NumberFormatException e) {
            // System.out.println("Input String cannot be parsed to Integer.");
        }
        return false;
    }

    public static boolean isFloat(String string) {
        float intValue;

        // System.out.println(String.format("Parsing string: \"%s\"", string));

        try {
            intValue = Float.parseFloat(string);
            return true;
        } catch (NumberFormatException e) {
            // System.out.println("Input String cannot be parsed to Integer.");
        }
        return false;
    }
}

File: TP 1/Q7(DONE 99_)/tp1.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

int main()
{
    char word[1000];
    char x1[3], x2[3], x3[3], x4[3];

    do
    {
        scanf(" %[^\n]", word);

        if (strcmp(word, "FIM") != 0) // Se entrada for diferente de FIM, a execucao continua
        {

            if (isConsonant(word) == true)
            {
                printf("SIM ");
            }
            else
            {
                printf("NAO ");
            }
            if (isVogal(word) == true)
            {
                printf("SIM ");
            }
            else
            {
                printf("NAO ");
            }
            if (isInteger(word) == true)
            {
                printf("SIM ");
            }
            else
            {
                printf("NAO ");
            }
            if (isFloat(word) == true)
            {
                printf("SIM\n");
            }
            else
            {
                printf("NAO\n");
            }
            // printf("%s %s %s %s", &x1, &x2, &x3, &x4);
        }
    } while (strcmp(word, "FIM") != 0);

    return 0;
}

int isConsonant(char word[])
{
    bool result = true;
    for (int index = 0; index < strlen(word); index++)
    {
        char charat = word[index];
        if (!(charat != 'a' && charat != 'e' && charat != 'i' && charat != 'o' && charat != 'u' && charat != 'A' 
        && charat != 'E' && charat != 'I' && charat != 'O' && charat != 'U' && charat != '0' 
        && charat != '1' && charat != '2' && charat != '3' && charat != '4' && charat != '5' 
        && charat != '6' && charat != '7' && charat != '8' && charat != '9'))
        {
            result = false;
        }
    }
    return result;
}

int isVogal(char word[])
{
    bool result = true;
    for (int index = 0; index < strlen(word); index++)
    {
        char charat = word[index];
        // char charat = charat;
        if (charat != 'a' && charat != 'e' && charat != 'i' && charat != 'o' && charat != 'u' && charat != 'A' && charat != 'E' && charat != 'I' && charat != 'O' && charat != 'U')
        {
            result = false;
        }
    }
    return result;
}

int isInteger(char word[])
{
    bool result = true;
    for (int index = 0; index < strlen(word); index++)
    {
        char charat = word[index];
        if (charat != '0' && charat != '1' && charat != '2' 
        && charat != '3' && charat != '4' && charat != '5' 
        && charat != '6' && charat != '7' && charat != '8' && charat != '9')
        {
            result = false;
        }
    }
    return result;
}

int isFloat(char word[])
{
    bool hasPoint = false;
    bool hasNumber = false;
    bool hasChar = false;
    int length = strlen(word);
    for (int index = 0; index < length; index++)
    {
        char charat = word[index];
        if (!((charat >= '0') && (charat <= '9')))
        {
            if (charat == '.')
            {
                hasPoint = true;
            }
        }
        else if ((charat >= '0') && (charat <= '9'))
        {
            hasNumber = true;
        }
        else if (isConsonant(word) == true || isVogal(word) == true)
        {
            hasChar = true;
        }
    }
    return (hasPoint && hasNumber && !hasChar);
}